src/cargo/util/progress.rs

```diff
@@ -127,12 +127,25 @@ impl<'cfg> State<'cfg> {
 
         // Write out a pretty header, then the progress bar itself, and then
         // return back to the beginning of the line for the next print.
+        self.try_update_max_width();
         if let Some(string) = self.format.progress_status(cur, max, msg) {
             self.config.shell().status_header(&self.name)?;
             write!(self.config.shell().err(), "{}\r", string)?;
         }
         Ok(())
     }
+
+    fn clear(&mut self) {
+        self.try_update_max_width();
```

Comment:
Shouldn't we use *previous* width here? I am thinking about the following sequence:

1) we print 80col line
2) user resizes terminal to 40 col, so the line wraps 
3) clear rubs-out only 40 cols, so we get both a wrapping and a tail of old line on the screen.

Reply:
Nah at least in my testing the current width is what you always want I think. The behavior I saw is that if you had a 10-character wide window with text that looked like:

```
0123456789
```

then if you resize it to 9 characters you get:

```
        0
123456789
```

In that when you make the window smaller it pushes the contents at the start of the line to the previous line. It certainly makes everything look funky, but your cursor is always placed at the very last line and printing the terminal's width always clears just what the terminal has